app/models/identity.rb

```diff
@@ -29,6 +32,14 @@ def self.find_or_create_from_oauth(auth_hash)
     identity # return identity
   end
 
+  def unlink
+    if self.user.identities.length == 1 && !self.user.own_password
+      self.user.destroy
```

Comment:
Yes, the before destory action in the user modell destroys all conected identities.